dapp/src/services/IotaNamesIndexerClient.ts

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) 2025 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+export class IotaNamesIndexerClient {
+    private host: string;
+
+    constructor(host: string) {
+        this.host = host.endsWith('/') ? host.slice(0, -1) : host;
```

Comment:
nit: simpler way to extract the host
```suggestion
        this.host = new URL(host).origin;